### PR TITLE
SOLR-15124: Fix ThreadDumpHandlerTest

### PR DESCRIPTION
**Pull request by NazerkeBS**
https://issues.apache.org/jira/browse/SOLR-15124

# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java [hunk 1]
```diff
@@ -245,21 +253,18 @@ public void doTestOwnableSync(final boolean checkWaitingThreadViaPolling) throws
...
-  private NamedList<Object> readProperties() throws Exception {
+  private void request(String path, Consumer<NamedList<?>> consumer) throws Exception {
     SolrClient client = new EmbeddedSolrServer(h.getCore());
-
-    NamedList<Object> properties = client.request(new GenericSolrRequest(SolrRequest.METHOD.GET, "/admin/info/threads",
-            new ModifiableSolrParams()));
-    return properties;
+    ModifiableSolrParams mparams = new ModifiableSolrParams();
+    mparams.set("indent", true);
+    NamedList<?> req = client.request(new GenericSolrRequest(SolrRequest.METHOD.GET, path, mparams));
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java:260]: nit: would this be response, not request?

Comment (NazerkeBS) [reply solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java]: yess, it should be response

File: solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java [hunk 2]
```diff
@@ -245,21 +253,18 @@ public void doTestOwnableSync(final boolean checkWaitingThreadViaPolling) throws
...
+    consumer.accept(rsp);
   }
 
-  @SuppressWarnings({"unchecked"})
-  private NamedList<Object> getThread(NamedList<Object> threads, String threadName) {
-    for (Map.Entry<String, Object> threadEntry : threads) {
-      @SuppressWarnings({"unchecked"})
-      NamedList<Object> thread = (NamedList<Object>) threadEntry.getValue();
-      if (thread._getStr("name",  null).contains(threadName)) {
+  private NamedList<?> getThreadInfo(NamedList<?> req, String threadName) {
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java:264]: The first arg is the *response* not a *request*

**Review by dsmiley — APPROVED** [review]
Much better.  Just one minor parameter name issue.

File: solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java [hunk 3]
```diff
@@ -245,21 +253,18 @@ public void doTestOwnableSync(final boolean checkWaitingThreadViaPolling) throws
...
+    for (Map.Entry<String, ?> threadInfoEntry : (NamedList<?>) req._get("system/threadDump", null)) {
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java:265]:
```suggestion
  private NamedList<?> getThreadInfo(NamedList<?> rsp, String threadName) {
    for (Map.Entry<String, ?> threadInfoEntry : (NamedList<?>) rsp._get("system/threadDump", null)) {
```

**Pull request merged**